workflow/controller/cache/configmap_cache.go

```diff
@@ -61,6 +66,24 @@ func (c *configMapCache) validateConfigmap(ctx context.Context, cm *apiv1.Config
 }
 
 func (c *configMapCache) Load(ctx context.Context, key string) (*Entry, error) {
+	var entry *Entry
+	err := retry.OnError(kwait.Backoff{
+		Duration: time.Second,
+		Factor:   2,
+		Jitter:   0.1,
+		Steps:    5,
+		Cap:      30 * time.Second,
+	}, func(err error) bool {
+		return argoerr.IsTransientErr(ctx, err) || (apierr.IsConflict(err) && strings.Contains(err.Error(), "Operation cannot be fulfilled on configmaps"))
```

Comment:
Properly reviewing this now, why would other IsConflicts not want a retry - I'd prefer we drop the fallible and possibly breaking in the future string match here and in the other place and just check `IsConflict`